File: 0151_reverse_words_in_a_string/Solution.java
class Solution {
    public String reverseWords(String s) {
        String phase = s.trim().replaceAll(" +", " ");

        String[] words = phase.split(" ");
        String[] reverseWords = new String[words.length];

        for (int x = words.length - 1, z = 0; x >= 0; x--, z++) {
            reverseWords[z] = words[x];
        }

        String reversePhase = String.join(" ", reverseWords);

        return reversePhase;
    }
}

File: 0151_reverse_words_in_a_string/Solution.js
var reverseWords = function (s) {
    const phase = s.replace(/  +/g, ' ').trim();
    const arr = phase.split(" ");
    const revArr = arr.toReversed();
    const revPhase = revArr.join(" ");

    return revPhase;
};

// return (((s.replace( /  +/g, ' ' ).trim()).split(" ")).toReversed()).join(" ");
